### DISCUSSION TITLE
When exception return screenshot

### DISCUSSION THREAD

**Discussion by Devilmaydante** [MontFerret/ferret#784] [Q&A] [answered]
Hello dear community,

I want to know if today it's possible to return a SCREENSHOT when for example we have an error on a WAIT_ELEMENT or another command which is blocking?

This usecase is more to detect a potential change on the website and get directly a screenshot to do a visual check.

Thank you everyone and in particulary @ziflex for your work ❤️

**Comment by ziflex** [comment, marked as answer]
Hey,

Thanks for your question, it's quite a good one! 
At this moment, the only thing I can think of is something like this:

```aql
LET doc = document(@url)
LET ok = wait_element(doc, "#my-element")?
RETURN ok ? (
  FOR el IN element(doc, "#my-element").children
      RETURN el
  ) : ({
      error: true,
      screenshot: screenshot(doc)
  })
```

**Reply by Devilmaydante** [reply]
Thx Ziflex for your answer, I will dig how I can implement it with your tricks 👍 
For the future if you have an idea to manage the exceptions throwing you will be my hero 🥇
